sklearn/calibration.py

```diff
@@ -33,25 +33,36 @@
 from .utils.validation import _deprecate_positional_args
 
 
-def _fit_calibrated_classifer(estimator, X, y, train, test, supports_sw,
-                              method, classes, sample_weight=None):
+def _fit_calibrated_classifer(estimator, X, y, train, test,
+                              method, classes, sample_weight=None,
+                              sample_weight_base_estimator=None, **fit_params):
     """Fit calibrated classifier for a given dataset split.
 
     Returns
     -------
     calibrated_classifier : estimator object
         The calibrated estimator.
     """
-    if sample_weight is not None and supports_sw:
+    fit_params_train = {}
+    for key, val in fit_params.items():
+        check_consistent_length(y, val)
```

Comment:
Can we do the `check_consistent_length` all at once in `fit`? This way the parallel calls do not need to validate anymore.